build_tools/cmake/iree_check_test.cmake

```diff
@@ -268,10 +282,52 @@ function(iree_check_single_backend_test_suite)
         ${_RULE_OPT_TOOL}
       OPT_FLAGS
         ${_RULE_OPT_FLAGS}
+      TARGET_CPU_FEATURES
+        ${_RULE_TARGET_CPU_FEATURES}
     )
   endforeach()
 endfunction()
 
+# Helper function parsing a string occurring as an entry in TARGET_CPU_FEATURES_VARIANTS.
+#
+# This function has 3 output-params: variables that it sets with PARENT_SCOPE:
+# _ENABLED, _TARGET_CPU_FEATURES, _TARGET_CPU_FEATURES_SUFFIX.
+#
+# The special value "default" is handled here, resulting in the test always being enabled, with no target CPU feature flags and no suffix.
+#
+# Other values are parsed as "arch:features", the parsed arch is matched with CMAKE_SYSTEM_PROCESSOR, the test is enabled
+# if and only if there is a match, and then the parsed features are passed as target CPU featues, and the test suffix
+# is set accordingly.
```

Comment:
I think an example here would be very helpful. Also this refers to some of the outputs in somewhat imprecise terms that I assume come from its expected usage or prior existence as a comment block within a rule. Plus a few typos and line wrapping


```suggestion
# "default" is handled specially. _ENABLED is always set to "TRUE" and
# _TARGET_CPU_FEATURES and _TARGET_CPU_FEATURES_SUFFIX are both set to the
# empty string.
#
# Other values are parsed as "arch:features", the parsed arch is matched with
# `CMAKE_SYSTEM_PROCESSOR`, `_ENABLED` is set to "TRUE" if and only if they
# match, and `_TARGET_CPU_FEATURES_SUFFIX` is set to a string based on the
# features that is appropriate to include in a CMake target or test name. More
# than one target cpu feature is currently unsupported.

# aarch64:+dotprod -> _TARGET_CPU_FEATURES="+dotprod", _TARGET_CPU_FEATURES_SUFFIX="_dotprod"

# default -> _TARGET_CPU_FEATURES="", _TARGET_CPU_FEATURES_SUFFIX="", ENABLED="TRUE"
```

Reply:
done, thanks!